src/backend/cuda/JIT/BinaryNode.hpp

```diff
@@ -20,68 +20,34 @@ namespace JIT
     class BinaryNode : public Node
     {
     private:
-        const std::string m_op_str;
-        const int m_op;
-        const int m_call_type;
+        std::string m_op_str;
+        int m_op;
 
     public:
         BinaryNode(const char *out_type_str, const char *name_str,
-                   const std::string &op_str,
-                   Node_ptr lhs, Node_ptr rhs, int op, int call_type)
+                   const char *op_str,
+                   Node_ptr lhs, Node_ptr rhs, int op)
             : Node(out_type_str, name_str, std::max(lhs->getHeight(), rhs->getHeight()) + 1, {lhs, rhs}),
               m_op_str(op_str),
-              m_op(op),
-              m_call_type(call_type)
+              m_op(op)
         {
         }
 
         void genKerName(std::stringstream &kerStream, Node_ids ids)
         {
-            // Make the hex representation of enum part of the Kernel name
-            kerStream << "_" << std::setw(2) << std::setfill('0') << std::hex << m_op;
-            kerStream << std::setw(2) << std::setfill('0') << std::hex << ids.child_ids[0];
-            kerStream << std::setw(2) << std::setfill('0') << std::hex << ids.child_ids[1];
-            kerStream << std::setw(2) << std::setfill('0') << std::hex << ids.id << std::dec;
+            // Make the dec representation of enum part of the Kernel name
```

Comment:
I'll do another pass about reorganizing the JIT nodes at a later point (including trying to use the same code for CUDA and OpenCL JIT). Can we leave this be for now?